### PR TITLE
refactor: vulnerability plugin contract to accept client.Object 

### PR DESCRIPTION
**Pull request by deven0t**
This is needed for future development where we need more information on the object which we are scanning.

### CODE REVIEW

File: pkg/vulnerabilityreport/plugin.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package vulnerabilityreport
 import (
 	"io"
 
+	"sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (danielpacak) [inline pkg/vulnerabilityreport/plugin.go:6]: Use goimport without adding new lines between imported packages:

```go
import (
	"io"

	"github.com/aquasecurity/starboard/pkg/apis/aquasecurity/v1alpha1"
	"github.com/aquasecurity/starboard/pkg/docker"
	"github.com/aquasecurity/starboard/pkg/starboard"
	corev1 "k8s.io/api/core/v1"
	"sigs.k8s.io/controller-runtime/pkg/client"
)
```

File: pkg/vulnerabilityreport/plugin.go [hunk 2]
```diff
@@ -23,7 +25,7 @@ type Plugin interface {
 	// The second argument maps container names to Docker registry credentials,
 	// which can be passed to the scanner as environment variables with values
 	// set from returned secrets.
-	GetScanJobSpec(ctx starboard.PluginContext, spec corev1.PodSpec, credentials map[string]docker.Auth) (
+	GetScanJobSpec(ctx starboard.PluginContext, obj client.Object, credentials map[string]docker.Auth) (
```
Comment (danielpacak) [inline pkg/vulnerabilityreport/plugin.go:28]: To indicate that we expect K8s workloads, i.e. object that do have PodSpec template:
```suggestion
	GetScanJobSpec(ctx starboard.PluginContext, workload client.Object, credentials map[string]docker.Auth) (
```

File: pkg/vulnerabilityreport/builder_test.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 	"testing"
 	"time"
 
+	"sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (danielpacak) [inline pkg/vulnerabilityreport/builder_test.go:8]: See my other comment about formatting with goimports and new lines.

File: pkg/plugin/trivy/plugin.go [hunk 4]
```diff
@@ -172,8 +174,9 @@ func (c Config) setResourceLimit(configKey string, k8sResourceList *corev1.Resou
 }
 
 type plugin struct {
-	clock       ext.Clock
-	idGenerator ext.IDGenerator
+	clock          ext.Clock
+	idGenerator    ext.IDGenerator
+	objectResolver *kube.ObjectResolver
```
Comment (danielpacak) [inline pkg/plugin/trivy/plugin.go:179]: I'd add it in the commit where we actually use it. Here, according to the change description, we are only changing the interface of vulnerability plugins.

File: pkg/plugin/trivy/plugin.go [hunk 5]
```diff
@@ -207,7 +211,12 @@ func (p *plugin) Init(ctx starboard.PluginContext) error {
 	})
 }
 
-func (p *plugin) GetScanJobSpec(ctx starboard.PluginContext, spec corev1.PodSpec, credentials map[string]docker.Auth) (corev1.PodSpec, []*corev1.Secret, error) {
+func (p *plugin) GetScanJobSpec(ctx starboard.PluginContext, obj client.Object, credentials map[string]docker.Auth) (corev1.PodSpec, []*corev1.Secret, error) {
```
Comment (danielpacak) [inline pkg/plugin/trivy/plugin.go:214]:
```suggestion
func (p *plugin) GetScanJobSpec(ctx starboard.PluginContext, workload client.Object, credentials map[string]docker.Auth) (corev1.PodSpec, []*corev1.Secret, error) {
```

**Review by danielpacak — CHANGES_REQUESTED** [review]
Hey @deven0t Looks good. Just left a few remarks. Make sure that you don't change import statements that much.

File: pkg/plugin/trivy/plugin.go (hunk 4 above)
Comment (deven0t) [reply pkg/plugin/trivy/plugin.go:179]: I thought the same thing during the change, but there was some dependent change needed in test cases as well so included it in this PR. Anyway reverting it, we will add it in another PR. thanks.

**Review by danielpacak — APPROVED** [review]

**Pull request merged**